File: drug-discovery-ui/src/services/structure.ts
export type StructureJSON = {
  elements: string[]
  positions: number[][]
  bonds: number[][]
  source?: string
  inchikey?: string
}

export async function fetchStructureJSON(query: string): Promise<StructureJSON> {
  const resp = await fetch(`/api/v1/structure?query=${encodeURIComponent(query)}&format=json`)
  if (!resp.ok) throw new Error(`HTTP ${resp.status}`)
  const data = await resp.json()
  if (!data.ok || !data.data) throw new Error(data.error || 'No data')
  return data.data as StructureJSON
}

File: drug-discovery-ui/backend/app/api/models/structure.py
from typing import List, Optional
from pydantic import BaseModel, Field

class StructureQuery(BaseModel):
    query: str = Field(..., description="Molecule identifier: name, SMILES, or InChIKey")
    format: Optional[str] = Field("json", description="json|sdf")

class StructureJSON(BaseModel):
    elements: List[str]
    positions: List[List[float]]  # [[x,y,z], ...]
    bonds: List[List[int]]  # [i,j,order]
    source: str = "pubchem"
    inchikey: Optional[str] = None

class StructureResponse(BaseModel):
    ok: bool
    format: str
    data: Optional[StructureJSON] = None
    sdf: Optional[str] = None
    error: Optional[str] = None
